drivers/include/vcnl40x0.h

```diff
@@ -0,0 +1,147 @@
+    VCNL40X0_AMBIENT_AVERAGE_16,                 /**< Ambient average measurement 16 */
+    VCNL40X0_AMBIENT_AVERAGE_32,                 /**< Ambient average measurement 32 */
+    VCNL40X0_AMBIENT_AVERAGE_64,                 /**< Ambient average measurement 64 */
+    VCNL40X0_AMBIENT_AVERAGE_128,                /**< Ambient average measurement 128 */
+} vcnl40x0_ambient_avg_t;
+
+/**
+ * @brief Ambient light measurement rate
+ */
+typedef enum {
+    VCNL40X0_AMBIENT_RATE_1 = 0,                 /**< Ambient measurement rate 1 */
+    VCNL40X0_AMBIENT_RATE_2,                     /**< Ambient measurement rate 2 */
+    VCNL40X0_AMBIENT_RATE_3,                     /**< Ambient measurement rate 3 */
+    VCNL40X0_AMBIENT_RATE_4,                     /**< Ambient measurement rate 4 */
+    VCNL40X0_AMBIENT_RATE_5,                     /**< Ambient measurement rate 5 */
+    VCNL40X0_AMBIENT_RATE_6,                     /**< Ambient measurement rate 6 */
+    VCNL40X0_AMBIENT_RATE_8,                     /**< Ambient measurement rate 8 */
+    VCNL40X0_AMBIENT_RATE_10,                    /**< Ambient measurement rate 10 */
+} vcnl40x0_ambient_rate_t;
+
+
+/**
+ * @brief   Status and error return codes
+ */
+enum {
+    VCNL40X0_OK = 0,                             /**< Everything was fine */
+    VCNL40X0_ERR_I2C,                            /**< Error initializing the I2C bus */
+    VCNL40X0_ERR_NODEV                           /**< Error wrong device */
+};
+
+/**
+ * @brief Device initialization parameters
+ */
+typedef struct {
+    i2c_t i2c_dev;                               /**< I2C device which is used */
+    uint8_t i2c_addr;                            /**< Address on I2C bus */
+    uint8_t led_current;                         /**< LED current */
+    vcnl40x0_proximity_rate_t proximity_rate;    /**< Proximity rate */
+    vcnl40x0_ambient_avg_t ambient_avg;          /**< Ambient light average*/
+    vcnl40x0_ambient_rate_t ambient_rate;        /**< Ambient light average*/
+} vcnl40x0_params_t;
+
+/**
+ * @brief Device descriptor for the VCNL40X0 sensor
+ */
+typedef struct {
+    vcnl40x0_params_t params;                    /**< Device parameters */
+} vcnl40x0_t;
+
+/**
+ * @brief Initialize the given VCNL40X0 device
+ *
+ * @param[out] dev          Initialized device descriptor of VCNL40X0 device
+ * @param[in]  params       The parameters for the VCNL40X0 device
+ *
+ * @return                  VCNL40X0_OK on success
+ * @return                  -VCNL40X0_ERR_I2C if given I2C is not enabled in board config
+ * @return                  -VCNL40X0_ERR_NODEV if not a vcnl40X0 device
+ */
+int vcnl40x0_init(vcnl40x0_t *dev, const vcnl40x0_params_t *params);
+
+/**
+ * @brief Read proximity value from the vcnl40X0 device
+ *
+ * @param[in] dev           Device descriptor of VCNL40X0 device to read from
+ *
+ * @return                  Proximity
+ */
+uint16_t vcnl40x0_read_proximity(vcnl40x0_t *dev);
```

Comment:
We already know that I2C communication might not be reliable, hence reads (or writes) are bound to failure. Thus, I'd recommend to provide a way to pass any errors to the application, which will not be possible using `uint16_t` here (and below). That said, it might be better to define the function as `int vcnl40x0_read_proximity(vcnl40x0_t *dev, uint16_t *val)` and use the return value for OK or error codes.